webapp/loggedin/surveyViews/categories/categoriesView.js

```diff
@@ -1,90 +1,110 @@
+import * as CategoryState from '../category/categoryState'
 
 const CategoriesView = (props) => {
-  const { categories, nodeDef, createCategory, deleteCategory, canSelect, readOnly, analysis, setNodeDefProp } = props
-  const selectedItemUuid = nodeDef && NodeDef.getCategoryUuid(nodeDef)
+  const { canSelect, onSelect, selectedItemUuid, onClose } = props
 
   const i18n = useI18n()
-  const history = useHistory()
   const dispatch = useDispatch()
+  const { pathname } = useLocation()
 
-  const onClose = nodeDef ? history.goBack : null
+  const inCategoriesPath = Boolean(matchPath(pathname, appModuleUri(designerModules.categories)))
+
+  const survey = useSelector(SurveyState.getSurvey)
+  const readOnly = !useAuthCanEditSurvey()
+  const editedCategory = useSelector(CategoryState.getCategoryForEdit)
+  const categories = R.pipe(
+    Survey.getCategoriesArray,
+    R.map((category) => ({
+      ...category,
+      usedByNodeDefs: !R.isEmpty(Survey.getNodeDefsByCategoryUuid(Category.getUuid(category))(survey)),
+    }))
+  )(survey)
 
   const canDeleteCategory = (category) =>
     category.usedByNodeDefs ? dispatch(NotificationActions.notifyInfo({ key: 'categoryEdit.cantBeDeleted' })) : true
 
+  const onAdd = async () => {
+    const category = await dispatch(CategoryActions.createCategory())
+    if (onSelect) {
+      onSelect(category)
+    }
+  }
+
   const onDelete = (category) =>
     dispatch(
       DialogConfirmActions.showDialogConfirm({
         key: 'categoryEdit.confirmDelete',
         params: { categoryName: Category.getName(category) || i18n.t('common.undefinedName') },
-        onOk: () => deleteCategory(category),
+        onOk: () => dispatch(CategoryActions.deleteCategory(category)),
       })
     )
 
   return (
-    <ItemsView
-      itemLabelFunction={(category) => Category.getName(category)}
-      itemLink={appModuleUri(analysis ? analysisModules.category : designerModules.category)}
-      items={categories}
-      selectedItemUuid={selectedItemUuid}
-      onAdd={createCategory}
-      canDelete={canDeleteCategory}
-      onDelete={onDelete}
-      canSelect={canSelect}
-      onSelect={(category) => setNodeDefProp(NodeDef.propKeys.categoryUuid, Category.getUuid(category))}
-      onClose={onClose}
-      readOnly={readOnly}
-    />
+    <>
+      <ItemsView
+        itemLabelFunction={Category.getName}
+        items={categories}
+        itemLink={inCategoriesPath ? appModuleUri(designerModules.category) : null}
+        selectedItemUuid={selectedItemUuid}
+        onAdd={onAdd}
+        onEdit={(category) =>
+          !inCategoriesPath && dispatch(CategoryActions.setCategoryForEdit(Category.getUuid(category)))
+        }
+        canDelete={canDeleteCategory}
+        onDelete={onDelete}
+        canSelect={canSelect}
+        onSelect={onSelect}
+        onClose={onClose}
+        readOnly={readOnly}
+      />
+      {editedCategory && (
+        <PanelRight
```

Comment:
is this shown when path is not categories ?

Reply:
this is shown only when path is not categories, otherwise the CategoryView component is rendered.